client/app/account/edit/component/EditAccountForm.tsx

```diff
@@ -0,0 +1,127 @@
+import { MdAccountCircle } from 'react-icons/md'
+import { useAccountEdit } from '@/(auth)/hooks/useAccountEdit'
+import {
+  Form,
+  FormControl,
+  FormField,
+  FormItem,
+  FormLabel,
+  FormMessage,
+  Input,
+  Button,
+} from '@/_components/ui'
+import { colorMap } from '@/_consts'
+import { UserType } from '@/_types'
+
+type EditAccountFormProps = {
+  user: UserType
+}
+const EditAccountForm = ({ user }: EditAccountFormProps) => {
+  const { form, onSubmit, isPending } = useAccountEdit(user)
+
+  const colorArray = Object.entries(colorMap).map((colorArray) => {
+    return {
+      value: colorArray[0],
+      color: colorArray[1],
+    }
+  })
+
+  const iconWatch = form.watch('icon')
+
+  return (
+    <Form {...form}>
+      <form onSubmit={onSubmit} className='mx-auto mt-8 w-full'>
+        <div className='space-y-5'>
+          <FormField
+            control={form.control}
+            name='icon'
+            render={() => (
+              <FormItem>
+                <FormControl>
+                  <div className='flex'>
+                    <div className='flex h-full columns-1 flex-col'>
+                      <FormLabel className=' font-semibold'>Icon</FormLabel>
+                      <MdAccountCircle
+                        className={`mt-12 text-4xl ${colorMap[user.icon as keyof typeof colorMap]}`}
```

Comment:
Oh can I ask you why you add  assertion `as`?

And maybe it's better to show dynamic color here based on a use click which icon color? If so, use `iconWatch`!

Reply:
Without `as`, TS throws me an error, so I need to use `as keyof typeof` to clarify. 
And I changed to `iconWatch`. This is better! 
7966dfa9c5f32421d86d3601de2266d7d6a78be2

Reply:
I see! You got a type error because the schema said it is a string, but you want to try to access the key of colorMap, which is a string literal. We usually don't want to use assertion because it means we override the typescript suggestion(error). So technically, it's better to change the schema and colorMap, but this is a private project, so it's okay this time!

Reply:
@hyejin-seok You can fix it in this way

```
export const colorMap: { [key: string]: string } = {
  RED: 'text-pink-700',
  ORANGE: 'text-orange-500',
  YELLOW: 'text-yellow-500',
  GREEN: 'text-green-500',
  BLUE: 'text-blue-500',
  INDIGO: 'text-indigo-500',
  VIOLET: 'text-violet-500',
  BLACK: 'text-slate-700',
  WHITE: 'text-slate-400',
}
```